### PR TITLE
Fixes /welcomebot preview teamname command not working.

### PR DESCRIPTION
**Pull request by maisnamrajusingh**
#### Summary
- Fixes issue of `/welcomebot preview teamname` command not working.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-welcomebot/issues/71

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs)
 	found := false
 	for _, message := range p.getWelcomeMessages() {
 		if message.TeamName == teamName {
-			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
+			if err := p.previewWelcomeMessage(args, *message); err != nil {
```
Comment (hanzei) [inline server/command.go:91]: While I think the behavior change is reasonable, this makes `teamName` not longer needed. Would you please remove it, also update the slash command auto complete and `README.md`?

Thanks!

**Review by hanzei — CHANGES_REQUESTED** [review]
The behavior change is fine, but it needs to be carried through whole code and the docs.

File: server/command.go (hunk 1 above)
Comment (maisnamrajusingh) [reply server/command.go:91]: Sure, Will do.

Comment (maisnamraju) [reply server/command.go:91]: @hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

Comment (maisnamrajusingh) [reply server/command.go:91]: @hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

Comment (hanzei) [reply server/command.go:91]: Would you mind including the docs and autocomplete changes in this PR? That makes it easier to review if a PR is complete-

Comment (jasonblais) [reply server/command.go:91]: cc @maisnamrajusingh on the above note from hanzei

Comment (maisnamrajusingh) [reply server/command.go:91]: got it guys.

Comment (maisnamrajusingh) [reply server/command.go:91]: @hanzei  I think, there  is a small confusion here. There is no need to change the docs in this case as we are still using the `teamName` to compare the config values that we get from the plugin config. Please check line no 90 on the same file.

Comment (hanzei) [reply server/command.go:91]: Could you please elaborate why `teamName` is check, if `previewWelcomeMessage` uses `args.TeamId` anyway?

Comment (maisnamrajusingh) [reply server/command.go:91]: @hanzei  the way the command works is `/welcomebot preview <TeamName>` so we need the team name to check which messages in the config relate to the particular team in the command.

Comment (hanzei) [reply server/command.go:91]: The team name passed to the command is not longer coupled with the `args.TeamId` used to post the preview. Hence a user could use `/welcomebot preview SomeOtherTeamName` and still get a preview for the current team.

Comment (maisnamrajusingh) [reply server/command.go:91]: @hanzei  sorry I thought this pr was merged, let me look into this today and get back to you

Comment (jasonblais) [reply server/command.go:91]: @maisnamrajusingh Any updates on this PR, or an ETA when it's ready for a review?

Comment (maisnamrajusingh) [reply server/command.go:91]: @hanzei  sorry, it took so long to respond to this. If you look at the code, you can see that the teamName is still used to make the comparison to check if the message belongs to that team. Or did I misunderstand your comment?

```
func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) {
	found := false
	for _, message := range p.getWelcomeMessages() {
		if message.TeamName == teamName {
			if err := p.previewWelcomeMessage(args, *message); err != nil {
				p.postCommandResponse(args, "error occurred while processing greeting for team `%s`: `%s`", teamName, err)
				return
			}

			found = true
		}
	}

	if !found {
		p.postCommandResponse(args, "team `%s` has not been found", teamName)
	}
}
```

Comment (jasonblais) [reply server/command.go:91]: cc @hanzei quick reminder on above

Comment (hanzei) [reply server/command.go:91]: The current behavior on `master` is: The user has to pass a team name, which is used to post the message.

With your PR the behavior is: The user has to pass a team name, which is used to find the welcome message, but the template itself is executed within the context of the current team.

Either the behavior should stay the same as in `master` or the team name should not longer be required and the current team is used.

Does that make things more clear?

Comment (maisnamraju) [reply server/command.go:91]: @hanzei makes sense. will update the logic and update accordingly.

**Comment by maisnamrajusingh** [comment]
@hanzei  I over thought what the problem was. The issue actually was that the user had `Cudy` as the team name and in the config whereas the `GetTeamByName` was expecting the values in lowercase. I reverted back my previous changes and added a string conversion to lower case when we fetch the team by its name

**Comment by maisnamrajusingh** [comment]
@hanzei  can you take a look at this and see if it needs anything else from my side ?

File: server/welcomebot.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func (p *Plugin) getSiteURL() string {
 	return *config.ServiceSettings.SiteURL
 }
 
-func (p *Plugin) newSampleMessageTemplate(teamName, userID string) (*MessageTemplate, error) {
+func (p *Plugin) newSampleMessageTemplate(teamName string, userID string) (*MessageTemplate, error) {
```
Comment (hanzei) [inline server/welcomebot.go:60]: Nit pick: Unrelated change

**Review by hanzei — APPROVED** [review]
Looks good to me :+1:

**Comment by hanzei** [comment]
@jfrerich Please take a look

**Review by jfrerich — APPROVED** [review]
LGTM!

**Comment by hanzei** [comment]
Deferring QA to release testing
